### ISSUE TITLE
Column -1 does not exist when using negative AddYears or AddMonths (PostgreSQL)

### ISSUE THREAD

**Issue by ghost** [linq2db/linq2db#3783]
### Describe your issue

In queries where there is AddYears or AddMonths with negative numbers.

For example (this query does not make sense but  it's just a simple query to demonstrate the issue):

`appDataConnection.Example
            .Where(x => x.date > x.date.Value.AddMonths(-1))
            .ToList();`

translates to:

`SELECT
	x.id,
	x.date
FROM
	example x
WHERE
	x.date > x.date + −1 * Interval '1 Year'`

This statement throws an exception **in some cultures, like 'eu'**, because the UNICODE used for the symbol "-" is different depending on the culture. For "en" is U+002D (it works) but for "eu" is U+2212 (throws exception).

Npgsql.PostgresException (0x80004005): 42703: column "−1" does not exist

POSITION: 103
   at Npgsql.Internal.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|215_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader(CommandBehavior behavior)
   at LinqToDB.Data.DataConnection.ExecuteReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteDataReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteReader()
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReader()
   at LinqToDB.Linq.QueryRunner.ExecuteQuery[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber)+MoveNext()
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at LinqToDB.Linq.Builder.FirstSingleBuilder.FirstSingleContext.<>c__DisplayClass3_0`1.<GetFirstElement>b__0(IDataContext db, Expression expr, Object[] ps, Object[] preambles)
   at LinqToDB.Linq.ExpressionQuery`1.System.Linq.IQueryProvider.Execute[TResult](Expression expression)
   at System.Linq.Queryable.First[TSource](IQueryable`1 source)
   at ConsoleApp1.Program.Main(String[] args) in /Users/scardenas/Development/ConsoleApp1/ConsoleApp1/Program.cs:line 36
  Exception data:
    Severity: ERROR
    SqlState: 42703
    MessageText: column "−1" does not exist
    Hint: Perhaps you meant to reference the column "x.id".
    Position: 103
    File: parse_relation.c
    Line: 3294
    Routine: errorMissingColumn

### Steps to reproduce

```c#
using System.Globalization;
using LinqToDB;
using LinqToDB.AspNet;
using LinqToDB.Configuration;
using LinqToDB.Data;
using LinqToDB.Mapping;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace ConsoleApp1;

public class Program
{
    public static void Main(string[] args)
    {
        var host = new HostBuilder()
            .ConfigureServices(services =>
            {
                services.AddLinqToDBContext<AppDataConnection>((provider, options) =>
                {
                    options.UsePostgreSQL("connection string");
                });
            })
            .UseConsoleLifetime()
            .Build();

        Thread.CurrentThread.CurrentCulture = new CultureInfo("eu");
        Thread.CurrentThread.CurrentUICulture = new CultureInfo("eu");
        
        var appDataConnection = host.Services.GetRequiredService<AppDataConnection>();
        var sql = appDataConnection.Example
            .Where(x => x.date > x.date.Value.AddYears(-1))
            .ToString();

        var res = appDataConnection.Example
            .Where(x => x.date > x.date.Value.AddYears(-1))
            .First();
        
        Console.WriteLine(sql);
    }
}

public class AppDataConnection: DataConnection
{
    public AppDataConnection(LinqToDBConnectionOptions<AppDataConnection> options)
        :base(options)
    {

    }

    public ITable<Example> Example => this.GetTable<Example>();
}

public class Example
{
    [PrimaryKey]
    public int id { get; set; }
    public DateTime? date { get; set; }
}
```

### Environment details
`Linq To DB` version: 4.2

Database (with version): PostgreSQL 11.12

ADO.NET Provider (with version): NpgSql 6.0.7

Operating system: MacOs & Windows

.NET Version: 6.0.401

**Comment by MaceWindu** [comment]
We also should add tests for negative values to ensure all providers handle it properly

**Comment by MaceWindu** [comment]
This should be fixed with #4380
